elfo-logger/src/actor.rs

```diff
@@ -120,61 +129,81 @@ impl Logger {
         }
     }
 
-    pub(super) fn format_event<T: theme::Theme>(&mut self, event: PreparedEvent) {
-        let out = &mut self.buffer;
+    fn format_event(&mut self, use_colors: bool, event: PreparedEvent) {
+        // boolean operator || is short-circuit
+        let successful = if use_colors {
+            self.do_format_event::<theme::ColoredTheme, FailOnUnfit>(&event)
+                || self.do_format_event::<theme::ColoredTheme, TruncateOnUnfit>(&event)
+        } else {
+            self.do_format_event::<theme::PlainTheme, FailOnUnfit>(&event)
+                || self.do_format_event::<theme::PlainTheme, TruncateOnUnfit>(&event)
+        };
+
+        if !successful {
+            unreachable!("truncation must succeed")
+        }
+    }
+
+    fn do_format_event<T: theme::Theme, F: LineFactory>(&mut self, event: &PreparedEvent) -> bool {
         let config = self.ctx.config();
+        let mut line = F::create_line(&mut self.buffer);
 
         let payload = self
             .shared
             .pool
             .get(event.payload_id)
             .expect("unknown string");
-        self.shared.pool.clear(event.payload_id);
 
         // <timestamp> <level> [<trace_id>] <object> - <message>\t<fields>
 
-        T::Timestamp::fmt(out, &event.timestamp);
-        out.push(' ');
-        T::Level::fmt(out, event.metadata.level());
-        out.push_str(" [");
-        T::TraceId::fmt(out, &event.trace_id);
-        out.push_str("] ");
-        T::ActorMeta::fmt(out, &event.object);
-        out.push_str(" - ");
-        T::Payload::fmt(out, &payload);
+        T::Timestamp::fmt(line.meta_mut(), &event.timestamp);
+        line.meta_mut().push(' ');
+        T::Level::fmt(line.meta_mut(), event.metadata.level());
+        line.meta_mut().push_str(" [");
+        T::TraceId::fmt(line.meta_mut(), &event.trace_id);
+        line.meta_mut().push_str("] ");
+        T::ActorMeta::fmt(line.payload_mut(), &event.object);
```

Comment:
Why not `meta_mut` here?

Reply:
Since `event.object` and `payload` are part of the payload, originally